src/main/java/com/conveyal/datatools/manager/models/transform/PreserveCustomFieldsTransformation.java

```diff
@@ -62,15 +62,17 @@ private static HashMap<String, Map<String, String>> createCsvHashMap(CsvMapReade
     public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status status) {
         String tableName = table + ".txt";
         Path targetZipPath = Paths.get(zipTarget.gtfsFile.getAbsolutePath());
-
         Table specTable = Arrays.stream(Table.tablesInOrder)
-            .filter(t -> t.name.equals(table))
-            .findFirst()
-            .get();
-        List<String> specTableFields = specTable.specFields().stream().map(f -> f.name).collect(Collectors.toList());
-        List<String> tablePrimaryKeys = specTable.getPrimaryKeyNames();
+                .filter(t -> t.name.equals(table))
+                .findFirst()
+                .get();
+
+        try (FileSystem targetZipFs = FileSystems.newFileSystem(targetZipPath, (ClassLoader) null)) {
+            if (specTable == null) throw new Exception(String.format("could not find specTable for table %s", table));
```

Comment:
Move the check outside of the try block.

Reply:
Actually, the more proper thing to check is `.isPresent()` on the result of `.findFirst()` above. If `.isPresent()` is true, then you can safely call `.get()`.

Reply:
And move to make the method throw an exception if the result is not present?

Reply:
Something like this, above the try block: 

```suggestion
           Optional<Table> streamResult = Arrays.stream(Table.tablesInOrder)
                .filter(t -> t.name.equals(table))
                .findFirst();

        if (!streamResult.isPresent()) {throw new Exception(String.format("could not find specTable for table %s", table));}
        Table specTable = streamResult.get();
```

Reply:
Yes, please.

Reply:
(Note that, from the documentation, calling `.get()` when `.isPresent()` returns `false` also triggers an exception.)

Reply:
Updated in https://github.com/ibi-group/datatools-server/pull/536/commits/73ffc88304dd672662ef0922a67ed5aace2b1e62